local_planner/src/nodes/common.cpp

```diff
@@ -131,3 +131,55 @@ void normalize(geometry_msgs::Point &p) {
     p.z = 0;
   }
 }
+
+double velocitySigmoid(double max_vel, double min_vel, double slope,
+                       double v_old, double elapsed) {
+  max_vel += 0.05;
+  min_vel -= 0.05;
+  v_old -= min_vel;
+  double t_old = -1.0 / slope * log((max_vel - min_vel) / v_old - 1.0);
+  double t_new = t_old + elapsed;
+  double speed = min_vel + (max_vel - min_vel) / (1.0 + exp(-slope * t_new));
+  return speed;
+}
+
+double velocityLinear(double max_vel, double min_vel, double slope,
+                      double v_old, double elapsed) {
+  v_old -= min_vel;
+  double t_old = v_old / slope;
+  double t_new = t_old + elapsed;
+  double speed = min_vel + t_new * slope;
+  return speed;
+}
+
+double getAngularVelocity(double new_yaw, double curr_yaw) {
```

Comment:
If I understand it right, this function computes a good yaw rate to reach a given desired orientation and the current orientation, using some hard-coded p-gain. The function and parameter naming is a bit misleading: I could easily think that this derives an angular velocity based on two orientation samples. Also `new_yaw` makes it sound like an output. Maybe `desired_yaw` would more clearly indicate what the use of this quantity is.